hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java

```diff
@@ -39,13 +45,29 @@ public class ConsensusCreateTopicHandler implements TransactionHandler {
      *
      * @param txBody the {@link TransactionBody} with the transaction data
      * @param payer the {@link AccountID} of the payer
+     * @param keyLookup the {@link AccountKeyLookup} to use for key lookups
      * @return the {@link TransactionMetadata} with all information that needs to be passed to
      *     {@link #handle(TransactionMetadata)}
      * @throws NullPointerException if one of the arguments is {@code null}
      */
     public TransactionMetadata preHandle(
-            @NonNull final TransactionBody txBody, @NonNull final AccountID payer) {
-        throw new UnsupportedOperationException("Not implemented");
+            @NonNull final TransactionBody txBody,
+            @NonNull final AccountID payer,
+            @NonNull final AccountKeyLookup keyLookup) {
+        final var metaBuilder =
+                new SigTransactionMetadataBuilder(keyLookup).txnBody(txBody).payerKeyFor(payer);
+
+        final var op = txBody.getConsensusCreateTopic();
+        final var adminKey = asHederaKey(op.getAdminKey());
+        final var submitKey = asHederaKey(op.getSubmitKey());
+        if (adminKey.isPresent() || submitKey.isPresent()) {
```

Comment:
I think you can use `metaBuilder.addToReqNonPayerKeys` to which you can add `HederaKey` directly. So, we can avoid creating `ArrayList` here

Reply:
Good thought!
```suggestion
        if (adminKey.isPresent() || submitKey.isPresent()) {
```